app/graphql/types/interfaces/rules_preload.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Class to deal with preloading various attributes for rules, such as
+# rule results, compliant,and references
+module RulesPreload
+  include GraphQL::Schema::Interface
+
+  def rules(args = {})
+    context[:parent_profile_id] ||= {}
+    return all_rules if system_id(args).blank?
+
+    latest_test_result_batch(args).then do |latest_test_result|
```

Comment:
The whole idea here is not only to batch the results, but also to pre-set all of the rule related fields whenever the profile is found